tools/serverpod_cli/test/generator/dart/server_code_generator/custom_class_test.dart

```diff
@@ -68,5 +69,55 @@ void main() {
             'The fromJson method should pass data as dynamic to CustomClass.fromJson but doesn\'t.',
       );
     });
+
+    test(
+      'then toJsonForProtocol method should correctly serialize customClassField by calling the appropriate toJson method',
+      () {
+        var toJsonForProtocolMethod =
+            CompilationUnitHelpers.tryFindMethodDeclaration(
+          maybeClassNamedExample!,
+          name: 'toJsonForProtocol',
+        );
+
+        var toJsonForProtocolCode = toJsonForProtocolMethod!.toSource();
+
+        // This regex checks for this pattern:
```

Comment:
Suggestion: `// This regex checks for this pattern but allows any number on the import.`